packages/client/components/EmailPasswordAuthForm.tsx

```diff
@@ -80,8 +80,8 @@ const validateEmail = (email: string) => {
 const validatePassword = (password: string) => {
   return new Legitity(password)
     .required('Please enter a password')
-    .min(6, '6 character minimum')
     .max(1000, `That's a book, not a password`)
+    .matches(new RegExp("^(?=.*\\d)(?=.*[A-Z])(?=.*[a-z])(?=.*[^\\w\\d\\s:])([^\\s]){8,}$"), 'Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character')
```

Comment:
I think it's nicer to split off the minimal length. This simplifies the regex and gives a better user experience. I would keep the rest together though, so the user can see all these requirements at once. The length should also be checked after the special characters: `required(...).matches(...).min(...).max(...)`

I'm also a bit concerned as it creates additional friction for signups. Especially with these more complicated requirements it's much nicer if the user can see them before trying to enter something invalid. Also what is considered special characters is handled differently from site to site.
I think this deserves its own ticket to improve the UI.

Reply:
I changed this to break out the length using .min set to 8, but made everything else a grouped regex, so you get the error of what else is required all at once, instead of doing multiple messages based on each match
![image](https://user-images.githubusercontent.com/71724289/143063941-ec32cdd0-4d6b-4d52-beda-d6baa16e081a.png)
![image](https://user-images.githubusercontent.com/71724289/143063997-46844850-cc3d-4c7f-8180-33f7ecb47b50.png)

Reply:
Additionally, I used any characters no present in the set for the match a-z, A-Z, 0-9 so they have to include something special. I didn't want to require a certain special. What are your thoughts @Dschoordsch ?

Reply:
Good enough I think. I still think putting the `match` first would be better, otherwise you come up with 8 characters only to discover you need to include a number and special character as well.